orders/models.py

```diff
@@ -0,0 +1,76 @@
+from django.db import models
+from django.utils import timezone
+from daycare.models import DayCare
+from dogowner.models import DogOwner
+
+
+class StatusOptions(models.TextChoices):
+    Pending = 'P', 'Pending'
+    Approved = 'A', 'Approved'
+    Canceled = 'C', 'Canceled'
+    Finished = 'F', 'Finished'
+    OnGoing = 'O', 'On Going'
+
+
+class Order(models.Model):
+    order_id = models.AutoField(primary_key=True)
+    dog_owner_id = models.ForeignKey(DogOwner, on_delete=models.CASCADE)
+    daycare_id = models.ForeignKey(DayCare, on_delete=models.CASCADE)
+    booking_date = models.DateField(auto_now=True)
+    approving_date = models.DateField(blank=True, null=True)
+    starting_date = models.DateField()
+    finishing_date = models.DateField()
+    canceling_date = models.DateField(blank=True, null=True)
+    total_price = models.IntegerField()
+    status = models.CharField(max_length=1,
+                              choices=StatusOptions.choices,
+                              default='P'
+                              )
+
+    def __str__(self):
+        return f"Dog owner ID: {self.dog_owner_id}, daycare ID: {self.daycare_id}, status: {self.status}"
+
+    @staticmethod
+    def create(dog_owner_id, daycare_id, starting_date, finishing_date, total_price):
+        new_order = Order(dog_owner_id=dog_owner_id,
+                          daycare_id=daycare_id,
+                          booking_date=timezone.now(),
+                          starting_date=starting_date,
+                          finishing_date=finishing_date,
+                          total_price=total_price,
+                          status=StatusOptions.Pending
+                          )
+
+        new_order.save()
+        return new_order
+
+    @staticmethod
+    def get_all_dog_owner_orders(user_id):
+        filter_args = {
+            'dog_owner_id': user_id,
+        }
+
+        return Order.objects.filter(**filter_args)
+
+    @staticmethod
+    def get_all_daycare_orders(user_id):
+        filter_args = {
+            'daycare_id': user_id,
+        }
+
+        return Order.objects.filter(**filter_args)
+
+    @staticmethod
+    def get_order_by_order_id(order_id):
+        return Order.objects.filter(order_id=order_id)
+
+    def update_status(self, new_status):
+        old_status = self.status
+
+        if new_status == StatusOptions.Canceled and old_status != StatusOptions.Canceled:
+            self.canceling_date = timezone.now()
+        elif new_status == StatusOptions.Approved and old_status != StatusOptions.Approved:
+            self.approving_date = timezone.now()
+
+        self.status = new_status
+        self.save()
```

Comment:
This method is only updating the status to Cancel/Approve status?
I think that all the updates of status methods updating should consist of more logic and we need to separate each method for each update status, for example:
`def cancel_order` if an order is canceling (it can be the dogowner canceling the order or the daycare), then we need to update the status to Canceled. 
tell me what u think 
**note**: i think we can implement those methods in the future on another PR

Reply:
> This method is only updating the status to Cancel/Approve status? 

Yes, because those are the only 2 statuses that are possible for updating order status.

> I think that all the updates of status methods updating should consist of more logic and we need to separate each method for each update status, for example: `def cancel_order` if an order is canceling (it can be the dogowner canceling the order or the daycare), then we need to update the status to Canceled. 

Maybe you're right, but I don't know if it's such a big difference.
Of course, changing the status of the order as a command will be easier, so maybe your way is better.
I'll think about it. Thank you!

Reply:
Dont forget to follow up :slightly_smiling_face:

Reply:
I think you should keep it as it is for the time being to avoid complications, but do open an issue to research further on it later.

Reply:
I've changed to Erez's advice. There are 2 methods in order for updating order's status - approve_order and cancel_order

Reply:
huh, ok, I thought Erez suggested something else entierly. I agree with the way it is right now.